routing/turns_tts_text.cpp

```diff
@@ -60,20 +64,280 @@ std::string GetTtsText::GetTurnNotification(Notification const & notification) c
   if (notification.m_useThenInsteadOfDistance && notification.m_turnDir == CarDirection::None)
     return {};
 
-  std::string const dirStr = GetTextById(GetDirectionTextId(notification));
   if (dirStr.empty())
     return {};
 
-  std::string distStr;
+  std::string thenStr;
   if (notification.m_useThenInsteadOfDistance)
-  {
-    distStr = GetTextById("then");
-    if (notification.m_distanceUnits > 0)
-      distStr = distStr + " " + GetTextById(GetDistanceTextId(notification));
-  }
-  else
+    thenStr = GetTextById("then") + " "; // add space only if needed
+
+  std::string distStr;
+  if (notification.m_distanceUnits > 0)
     distStr = GetTextById(GetDistanceTextId(notification));
-  return distStr + " " + dirStr;
+
+  if (!notification.m_nextStreet.empty()) {
+    // We're going to pronounce the street name.
+
+    // First, let's get rid of unpronounceable symbols.
+    std::string streetOut = notification.m_nextStreet;
+    // Semicolons are between destinations
+    // and pronounced more like commas.
+    std::replace( streetOut.begin(), streetOut.end(), ';', ',');
+    // Open brackets have no pronunciation analogue
+    std::replace( streetOut.begin(), streetOut.end(), '[', ' ');
+    // Closed brackets end a highway and introduce the rest
+    std::replace( streetOut.begin(), streetOut.end(), ']', ':');
+    // An angle bracket is currently used to represent "to" a place
+    // Ideally we'd add a translation for "to" or create full format
+    // strings for better i18n support, like "turn onto %s toward %s"
+    std::replace( streetOut.begin(), streetOut.end(), '>', ',');
+
+    // Replace any full-stop characters to make TTS flow better.
+    // Full stops are: . (Period) or 。 (East Asian) or । (Hindi)
+    strings::ReplaceLast(distStr, ".", "");
+    strings::ReplaceLast(distStr, "。", "");
+    strings::ReplaceLast(distStr, "।", "");
+
+    // If the turn direction with the key +_street exists for this locale, use it (like make_a_right_turn_street)
+    std::string dirStreetStr = GetTextById(dirKey+"_street");
+    if (!dirStreetStr.empty())
+      dirStr = std::move(dirStreetStr);
+
+    // Normally use "onto" for "turn right onto Main St"
+    std::string ontoStr = GetTextById("onto");
+
+    // If the nextStreet begins with [123]: we'll announce it as an exit number
+    std::regex re("^\\[.+\\]:");
+    std::smatch m;
+    if (std::regex_search(notification.m_nextStreet, m, re) && m.size() > 0) {
+      // Try to get a specific "take exit #" phrase and its associated "onto" phrase (if any)
+      std::string dirExitStr = GetTextById("take_exit_number");
+      if (!dirExitStr.empty()) {
+        dirStr = std::move(dirExitStr);
+        ontoStr = ""; // take_exit_number overwrites "onto"
+
+        /* // TODO: currently unused, consider removing
+        // If 'onto_exit_number' exists for this locale, use it instead of "onto"
+        // (for english, null is good: "take exit 543" instead of "take exit onto 543")
```

Comment:
Perhaps in the distant future, but for now Google TTS does a surprisingly good job of saying `take exit s"yezzhaytena@ s"yezde` which I have no idea what it means, but if I was traveling in the former USSR at least that would match the street signs I'm seeing more closely than "take exit leave at the convention" which is what Google Translate would otherwise do and is total nonsense.

In any case `onto_exit_number` has been mostly replaced by `take_exit_number` and no languages appear to need further strings in and among that (maybe in the distant future if we need `take_exit_number_off_highway` and `take_exit_number_leaving_roundabout` but it's too early to attempt that) so I will remove this.